### ISSUE TITLE
Routine update of dependencies after additional pinning

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2917]
Update Poetry managed dependencies after additional pinning: found to be required/advisable via draft PR #2918 investigations.

Includes additional pinning in pyproject.toml.
All the following were unpinned previously:
- django-oauth-toolkit = "~2.4" (Avoid potential breaking changes).
- gevent = "24.2.1" (Avoid AUTH_REQ_SASL_CONT without AUTH_REQ_SASL artifact).
- gunicorn = "~23" (was "*") (Pin to major version, precaution)

Refresh poetry.lock via poetry update.

```
cd /opt/rockstor
poetry update
```

Issue resolution via resulting updated pyproject.toml & poetry.lock files.

---

Upstream reference: https://python-poetry.org/docs/basic-usage/#updating-dependencies-to-their-latest-versions

**Comment by phillxnet** [comment]
# Resulting Poetry report
Leap 15.6 X86_64 host

```
Package operations: 0 installs, 20 updates, 1 removal

  • Removing pytz (2024.1)
  • Updating cffi (1.17.0 -> 1.17.1)
  • Updating cryptography (43.0.0 -> 43.0.1)
  • Updating more-itertools (10.4.0 -> 10.5.0)
  • Updating zipp (3.20.1 -> 3.20.2)
  • Updating certifi (2024.7.4 -> 2024.8.30)
  • Updating charset-normalizer (3.3.2 -> 3.4.0)
  • Updating idna (3.8 -> 3.10)
  • Updating importlib-metadata (8.4.0 -> 8.5.0)
  • Updating jaraco-functools (4.0.2 -> 4.1.0)
  • Updating setuptools (73.0.1 -> 75.2.0)
  • Updating simple-websocket (1.0.0 -> 1.1.0)
  • Updating urllib3 (2.2.2 -> 2.2.3)
  • Updating django (4.2.15 -> 4.2.16)
  • Updating greenlet (3.0.3 -> 3.1.1)
  • Updating keyring (25.3.0 -> 25.4.1)
  • Updating zope-interface (7.0.2 -> 7.1.0)
  • Updating django-oauth-toolkit (2.4.0 -> 3.0.1)
  • Updating gevent (24.2.1 -> 24.10.2)
  • Updating huey (2.5.1 -> 2.5.2)
  • Updating psycopg (3.2.1 -> 3.2.3)

Writing lock file
```

**Comment by phillxnet** [comment]
# Notable Changelog references:

- django: https://docs.djangoproject.com/en/5.0/releases/4.2.16/
- keyring: https://github.com/jaraco/keyring/blob/main/NEWS.rst
- gunicorn: https://docs.gunicorn.org/en/stable/news.html
- django-oauth-toolkit: https://django-oauth-toolkit.readthedocs.io/en/latest/changelog.html

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2919
